### ISSUE TITLE
New Contributor Self-Intro: Sukanta Roy

### ISSUE THREAD

**Issue by sukanta-27** [TEAMMATES/teammates#8154]
- **Link to my fork:** https://github.com/sukanta-27/teammates.git

**Screenshot of changes made (as seen on the UI)**

![screenshot 240](https://user-images.githubusercontent.com/10681755/32853460-5fb41c88-ca61-11e7-9b5d-5c06412c2d1d.png)

![screenshot 241](https://user-images.githubusercontent.com/10681755/32853481-6a063112-ca61-11e7-9cba-afb07f87dc52.png)

Added a new "Tagline" and changed the color of the key feature headers to orange. 

**Short self-introduction**

Hello TEAMMATES, My name is Sukanta Roy. I am a final year student of Information Technology.I want to contribute to TEAMMATES because of it's beginner friendly community as I am new to the Open Source Community and also because of it's well-organized workflow. I will also like to apply for Google Summer of Code 2018 with TEAMMATES once the Organization list for the 2018 GSoC is published and Student application opens. (So if there is any advice for me on how to start contributing in TEAMMATES, It will be very helpful.)

**Comment by chowyb** [comment]
Hi @sukanta-27,

Welcome to TEAMMATES! You may wish to start by taking on one of our `d.FirstTimers` issues, before moving on to `d.Contributors` issues :)

Also, while your interest in applying for GSoC with us is noted, I must warn you in advance that while we will quite likely be one of the organizations involved in GSoC, this is not guaranteed to be the case; that being said, we do still accept contributions outside of GSoC as well.

**Comment by sukanta-27** [comment]
@chowyb  Thank you very much.

**Comment by sukanta-27** [comment]
**The link to my app :**
http://teammates-sukanta.appspot.com/

**Comment by sukanta-27** [comment]
@chowyb Need some direction on how I can contribute:

I was looking for a `d.FirstTimers` issue to work on, but there are only 4 open issues.
out of which : 
#7706 is said to be complicated for a firstTimer, and a issue to address the root cause is opened in #7731 
<br>
and #8112 and #7350 already have open PRs
<br>
and in #7490 I am not sure whether any work is required or not as I agree with the point @wkurniawan07 made in the last comment, and because there is no activity in that issue after that.

I am not quite sure where to start. So, can you give me some directions on how to start contributing or what I can do meanwhile. Should I just go through the codebase and familiarize myself with the project more, and try to find a issue myself? Any help will be appreciated.

**Comment by wkurniawan07** [comment]
@sukanta-27 thanks for highlighting the current shortage of `d.FirstTimers` issues. I have just added a new one #8168 which you're free to work on.

**Comment by sukanta-27** [comment]
@wkurniawan07 Thank you very much :)

**Comment by sukanta-27** [comment]
@wkurniawan07 I don't know if mentioning my interest after the organization list for GSoC publishes is a necessary step or not, so thought of doing it anyway.

As I mentioned in my introduction I am aiming to be a part of GSoC 2018 with TEAMMATES. :smile:

**Comment by sukanta-27** [comment]
@wkurniawan07 @whipermr5 As you both are past GSoCers, wanted some advice for my proposal, if any one of you can respond, will be grateful. 

- Can writing proposal for more than one project increase my chance of getting selected, or should I just focus on one project only?

**Comment by damithc** [comment]
> Can writing proposal for more than one project increase my chance of getting selected, or should I just focus on one project only?

I guess it depends on the organization. For TEAMMATES, you can focus on one (or two, to be safe) projects, but also can mention in the application other projects you don't mind doing in your order of preference.

**Comment by sukanta-27** [comment]
@damithc Thank you sir. :smile:

**Comment by sukanta-27** [comment]
@damithc Sir, regarding the project proposal, should a applicant cover all the points in his/her proposal in some way or focusing on one/two points is enough? e.g. in the Questions+ project the features to consider are : 
>Features to consider:
>1. More configuration options for existing question types.
>2. More session templates containing sample questions (currently we have only two, and one of them is empty!)
>3. More stats and better visualization of responses for existing question types
>4. Improve the internal design/implementation of questions e.g. to make adding new question types easier, increase reuse, better testability

Here is covering every aspect necessary? 

Additionally should a applicant be confined in this domain only (here `f.Questions`) or can he/she propose works that slightly branch out to other domains as well (e.g. `f.Results` or `a.UIX` related issues)

**Comment by damithc** [comment]
> Here is covering every aspect necessary?

Not necessary. It's fine to focus on only _some_ of the items mentioned. 

> Additionally should a applicant be confined in this domain only (here f.Questions) or can he/she propose works that slightly branch out to other domains as well (e.g. f.Results or a.UIX related issues)

It's fine to branch out.

**Comment by sukanta-27** [comment]
@damithc @whipermr5 @wkurniawan07 
This may be a trivial question, but I was wondering if Teammates has a new feature/Update that it wants it's user base to know about, how does we do that? 

For example, in the context of Removing private sessions if we want to notify all the users that from now on we won't be supporting private sessions, how do we achieve that? Do we use the admin email feature to email all the registered users? 

**P.S:** Sorry for commenting on this issue, I try to use this issue for my general doubt clearing instead of creating a new issue.

**Comment by damithc** [comment]
> if Teammates has a new feature/Update that it wants it's user base to know about, how does we do that?

If it is a critical message, we have to send an email using the admin email feature. If not critical, we let users discover it themselves through usage. We started developing a _message of the day_ feature for this kind of use cases but we didn't manage to go beyond a primitive version of it, which is currently limited to students only.

**Comment by sukanta-27** [comment]
>If it is a critical message, we have to send an email using the admin email feature.

In this case, will it be a useful feature to have where in the `Admin Email` page, besides the `Upload` button, there will be a button (or a checkbox) which upon clicking will retrieve all the instructor email list, so that we can email all instructors to inform about any critical change? I know depending on what we consider "critical" this may not be used a lot, but I guess having a easier way to do this will help us in doing more of these updates (Like when we remove private sessions we should inform instructors that their sessions will become visible to other instructors in the course from a set point of time, or the fact that after #8687 we will support DST)? I focused on instructors as they are our primary users.

**Comment by damithc** [comment]
> I know depending on what we consider "critical" this may not be used a lot, but I guess having a easier way to do this will help us in doing more of these updates (Like when we remove private sessions we should inform instructors that their sessions will become visible to other instructors in the course from a set point of time, or the fact that after #8687 we will support DST)? I focused on instructors as they are our primary users.

While this can be useful, we also don't want to email *all* instructors too, as there are many instructors in the system who don't use TEAMMATES anymore. Who to email depends on the nature of the email. For example, only 20-30 instructors have private sessions; there is no need to email all 5000+ instructors about it.
